.github/workflows/docker.yml

```diff
@@ -26,10 +26,45 @@ jobs:
         with:
           context: .
           push: false
+          load: true
           cache-from: |
             type=gha,scope=main-${{ matrix.rust-features }}
             type=gha,scope=${{ github.ref_name }}-${{ matrix.rust-features }}
           cache-to: type=gha,scope=${{ github.ref_name }}-${{ matrix.rust-features }},mode=max
           build-args: |
             GIT_REVISION=${{ steps.git.outputs.GIT_REVISION }}
             RUST_FEATURES=${{ matrix.rust-features }}
+      # Test the dev compose, which should use the images built earlier. Technically this is only
+      # interesting when feature integration-testing is on, but we may as well exercise both.
+      - name: Compose (dev)
+        id: compose-dev
+        run: docker compose -f compose.dev.yaml up --wait --wait-timeout 120
+      - name: Inspect dev containers
+        if: ${{ failure() && steps.compose-dev.outcome != 'success' }}
+        run: |
+          docker compose ps
+          for NAME in `docker compose ps --format json | jq -r '.Name'`; do
+            docker inspect $NAME
+          done
+
+  # Test the non-dev compose, which we use for demo purposes. This pulls images from remote repos,
+  # so no need to build anything. Additionally we want coverage across multiple operating systems.
+  # Ideally we'd test on macOS, too, but those runners don't have Docker:
+  # https://github.com/actions/runner-images/issues/2150
+  compose:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
```

Comment:
Ah, I think this is missing `shell: bash`

Reply:
Yes, I'll try that but also I think the Docker in Windows images won't work:

> image operating system "linux" cannot be used on this platform: operating system is not supported